src/server/scripts/Commands/cs_premium.cpp

```diff
@@ -0,0 +1,221 @@
+
+    std::vector<ChatCommand> GetCommands() const override
+    {
+        static std::vector<ChatCommand> characterCommandTable
+        {
+            { "create",     SEC_GAMEMASTER, false, &HandlePremiumCharacterCreateCommand, "" },
+            { "delete",     SEC_GAMEMASTER, false, &HandlePremiumCharacterDeleteCommand, "" },
+            { "info",       SEC_GAMEMASTER, false, &HandlePremiumCharacterInfoCommand, "" }
+        };
+
+        static std::vector<ChatCommand> accountCommandTable
+        {
+            { "create",     SEC_GAMEMASTER, false, &HandlePremiumAccountCreateCommand, "" },
+            { "delete",     SEC_GAMEMASTER, false, &HandlePremiumAccountDeleteCommand, "" },
+            { "info",       SEC_GAMEMASTER, false, &HandlePremiumAccountInfoCommand, "" }
+        };
+        static std::vector<ChatCommand> commandTable =
+        {
+            { "account",    SEC_GAMEMASTER, false, nullptr, "", accountCommandTable},
+            { "character",  SEC_GAMEMASTER, false, nullptr, "", characterCommandTable}
+        };
+        static std::vector<ChatCommand> premiumCommandTable =
+        {
+            { "premium",    SEC_GAMEMASTER, false, nullptr, "", commandTable}
+        };
+        return premiumCommandTable;
+    }
+
+    static bool HandlePremiumCharacterInfoCommand(ChatHandler* handler, const char* /*args*/)
+    {
+        Player* target = handler->getSelectedPlayer();
+        string playerName = target->GetName().c_str();
+        if (!target)
+        {
+            (ChatHandler(handler->GetSession())).PSendSysMessage(LANG_NO_CHAR_SELECTED);
+            handler->SetSentErrorMessage(true);
+            return false;
+        }
+
+        if (handler->HasLowerSecurity(target, 0))
+            return false;
+
+        int premium_level = target->GetCharacterPremiumLevel(target->GetGUIDLow());
+        if (!premium_level)
+        {
+            (ChatHandler(handler->GetSession())).PSendSysMessage("No premium level available for character.");
+            handler->SetSentErrorMessage(true);
+            return false;
+        }
+        else
+            (ChatHandler(handler->GetSession())).PSendSysMessage("Character premium level: %u.", premium_level);
+
+        return true;
+    }
+
+    static bool HandlePremiumCharacterCreateCommand(ChatHandler* handler, const char* args)
+    {
+        if (!*args)
+            return false;
+
+        // Which premium level to be added
+        int premiumLevel = atoi((char*)args);
+
+        Player* target = handler->getSelectedPlayer();
+        string playerName = target->GetName().c_str();
+        if (!target)
+        {
+            handler->SendSysMessage(LANG_NO_CHAR_SELECTED);
+            handler->SetSentErrorMessage(true);
+            return false;
+        }
+
+        if (handler->HasLowerSecurity(target, 0))
+            return false;
+
+        bool characterPremiumLevelAdded = target->CreateCharacterPremiumLevel(target->GetGUIDLow(), premiumLevel);
```

Comment:
Reconsider the functions: You dont need to send the GUID as parameter, you already have "target" inside the function anyways

This would be a better function for this IMO.
`character->SetPremiumLevel(/*int8*/ level)`

Reply:
"Character granted premium level %u"

You dont "create" a character here, do you?

Reply:
I do need a create because my idea was you having to delete before creating, creates a layer of seccurity to avoid accidents. So you can't just set a premium level, before you have to delete and then create.